src/backend/distributed/planner/multi_planner.c

```diff
@@ -26,11 +26,14 @@
 #include "executor/executor.h"
 #include "nodes/makefuncs.h"
 #include "nodes/nodeFuncs.h"
+#include "parser/parsetree.h"
+#include "optimizer/pathnode.h"
 #include "optimizer/planner.h"
 #include "utils/memutils.h"
 
 
 static List *relationRestrictionContextList = NIL;
+static List *joinRestrictionContextList = NIL;
```

Comment:
Not really. `root->eq_classes` contains two separate `EQUIVALENCECLASS` es, which means that they are not known to be equal by postgres (and so we don't treat them being equal). 

However `JoinPathExtraData->restrictlist` contains an equality `OpExpr` among the corresponding `Vars`. 

That seems kind of weird to me as well. But, I couldn't figure out why this is happening. Do you have any ideas?